modin/experimental/core/execution/native/implementations/hdk_on_native/dataframe/dataframe.py

```diff
@@ -2556,3 +2566,60 @@ def _is_trivial_index(cls, index):
             and index.min() == 0
             and index.max() == len(index) - 1
         )
+
+
+class LazyProxyCategoricalDtype(pandas.CategoricalDtype):
+    """
+    Proxy class for lazily retrieving categorical dtypes from arrow tables.
+
+    Parameters
+    ----------
+    table : pyarrow.Table
+        Source table.
+    name : str
+        Column name.
+    """
+
+    def __init__(self, table: pyarrow.Table, name: str):
+        self._table = table
+        self._name = name
+        self._ordered = False
+        self._lazy_categories = None
+
+    def _new(self, table: pyarrow.Table, name: str) -> pandas.CategoricalDtype:
+        """
+        Create a new proxy, if either table or column name are different.
+
+        Parameters
+        ----------
+        table : pyarrow.Table
+            Source table.
+        name : str
+            Column name.
+
+        Returns
+        -------
+        pandas.CategoricalDtype
+        """
+        if self._table is None:
+            # The table has been materialized, we don't need a proxy anymore.
+            return pandas.CategoricalDtype(self.categories)
+        elif table is self._table and name == self._name:
+            return self
+        else:
+            return LazyProxyCategoricalDtype(table, name)
+
+    @property
+    def _categories(self):  # noqa: GL08
+        if self._table is not None:
+            chunks = self._table.column(self._name).chunks
+            cat = chunks[0].dictionary.to_pandas()
+            for i in range(1, len(chunks)):
+                cat = pandas.concat((cat, chunks[i].dictionary.to_pandas()))
```

Comment:
so there will be only a single concat rather than multiple ones
```suggestion
            cat = pandas.concat([chunk.dictionary.to_pandas() for chunk in chunks])
```